src/math/Bspline.cpp

```diff
@@ -28,27 +29,31 @@ Bspline::Bspline(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int splineDegree
   //The code for computing the knots and the control points is copied from Eigens bspline interpolation with minor modifications, https://gitlab.com/libeigen/eigen/-/blob/master/unsupported/Eigen/src/Splines/SplineFitting.h
 
   Eigen::KnotAveraging(ts, splineDegree, _knots);
-  Eigen::DenseIndex n = xs.cols();
-  Eigen::MatrixXd   A = Eigen::MatrixXd::Zero(n, n);
+  Eigen::DenseIndex                   n = xs.cols();
+  std::vector<Eigen::Triplet<double>> matrixEntries;
+  matrixEntries.reserve(n * splineDegree + 2);
 
   for (Eigen::DenseIndex i = 1; i < n - 1; ++i) {
-    //Attempt at hack... the spline dimension is not used here explicitly
-    const Eigen::DenseIndex span = Eigen::Spline<double, 1>::Span(ts[i], splineDegree, _knots);
+    const Eigen::DenseIndex span      = Eigen::Spline<double, 1>::Span(ts[i], splineDegree, _knots);
+    auto                    basisFunc = Eigen::Spline<double, 1>::BasisFunctions(ts[i], splineDegree, _knots);
 
-    // The segment call should somehow be told the spline order at compile time.
-    A.row(i).segment(span - splineDegree, splineDegree + 1) = Eigen::Spline<double, 1>::BasisFunctions(ts[i], splineDegree, _knots);
+    for (Eigen::DenseIndex j = 0; j < splineDegree + 1; ++j) {
+      matrixEntries.insert(matrixEntries.end(), Eigen::Triplet<double>(i, span - splineDegree + j, basisFunc(j)));
```

Comment:
My suggestion was the following:

```suggestion
      matrixEntries.emplace_back(i, span - splineDegree + j, basisFunc(j));
```

To clarify:

`emplace_back` constructs an element in-place using the given arguments. So, for a `vector` with element type `Eigen:Triplet<double>`, calling `emplace_back(x,y,z)` is like calling `push_back(Eigen::Triplet<double>(x,y,z))`.

In this case, this is mostly a readability improvement.

Reply:
Thanks for the nice explanation. I originally had 'push_back((x,y,z))' without the 'Eigen::Triplet<double>', which triggered an error in the linter so I switched to 'emplace_back((x,y,z))', since this was suggested by the linter.